storage/innobase/srv/srv0start.cc

```diff
@@ -2848,6 +2848,9 @@ innobase_start_or_create_for_mysql(void)
 	os_fast_mutex_free(&srv_os_test_mutex);
 
 	if (srv_rebuild_indexes) {
+		while (trx_rollback_or_clean_is_active) {
```

Comment:
It looks like that we might need proper synchronization here with a mutex or at least by making `trx_rollback_or_clean_is_active` volatile or by playing around with atomic.